docs/proposals/component-registration.md

```diff
@@ -0,0 +1,320 @@
+
+**ComponentStatuses** is a top-level Kubernetes API resource that exposes a `/componentstatuses` endpoint that shows component and Etcd status. It's unquestionably useful and used, but questionably named and implemented. However, it must continue to function (until v2) for reverse compatibility. See [Problems](#problems) for more details. `kubectl get componentstatuses` can be used from the CLI as a wrapper for the API..
+
+**Liveness** is a condition of a service instance that indicates it is running. When a local process is NotAlive it needs to be restarted. When a remote service is NotAlive it may been to be restarted, or the networking may need to be fixed.
+
+**Readiness** is a condition of a service instance that indicates it is Alive, able to handle new requests, and that its dependencies are also Ready. Ideally transitivity of readiness requires that only immediate dependencies be confirmed to be Ready. Readiness is usually expected to be easily assessed, slightly more expensive than liveness assessment, but still cheap enough to perform frequently and continuously. Readiness is not expected to provide robust metrics data, which may be more expensive to gather.
+
+**Canary Conditions** are conditions which indicate problems. Liveness and Readiness are canary conditions because when failing, they indicate problems, but do not guarantee health when not failing. Because of this distinction, and to avoid ambiguity, the term "health" should be avoided in this context.
+
+**Endpoints** is a namespaced Kubernetes API that describes a set of address:port pairs (with optional port name) that describe the providing instances of a Service. Currently only Ready addresses are included, but there is a PR to add NotReady addresses: #13778. Internal Endpoints are created and managed automatically by the Endpoints Controller. External Endpoints are created and managed manually by Kubernetes users.
+
+An **Internal Service** is a Service defined by Pod Selectors. Internal Endpoints for each Internal Service are created and managed automatically by the Endpoints Controller based on ServiceSpec Selectors.
+
+An **External Service** is a Service without Selectors. External Endpoints for each External Service are created and managed manually (or at least are not currently created/managed by primary Kubernetes components).
+
+
+## Abstract
+
+The goal of this proposal is to move Kubernetes a little closer to each of the following goals:
+
+- **High Availability**
+  - Allow multiple instances of each component
+- **Extensibility**
+  - Allow optional and third party components to be added to the cluster
+- **Deployment Flexibility**
+  - Allow components to be deployed on different machines and in different availability zones
```

Comment:
In this case it's the AZ the external components may be in, not the AZ the k8s nodes are in. Addon Components are obviously constrained to be on k8s nodes and implicitly in the same AZ. 

This makes more sense if you've heard the convo on the distinction between "the cluster" that k8s owns, and "the cluster" that k8s is IN. One includes external & primary components. The other doesn't.